core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -28,16 +28,23 @@ import org.apache.spark.internal.config
 import org.apache.spark.storage.BlockManagerId
 import org.apache.spark.util.Utils
 
+trait Location extends Externalizable {
+  def host: String
+  def port: Int
+  def hostPort: String
+  def executorId: String = "unknown"
```

Comment:
I added this only for the convenient purpose, doesn't mean I have any preference for the interface.

Reply:
Yeah, adding this here helps for this initial version. Could we add something like `storageInfo: Option[Serializable]`, which could be used to store extra information for different disaggregated shuffle solutions? e.g. storageInfo could be remote storage file path(s) or remote shuffle server(s).

Reply:
For "store extra information", I think implementors can add whatever they want only if they're serializable. e.g., extra info can be `Option[Serializable]` or `Map[String, String]`.

But I did think about adding a common `StorageType` class to `Location`. For example, a valid use case is that we could know from it whether the storage is reliable (e.g., `location.storageType.isReliable`) so we can decide whether to apply "decommission" on such storage.

Reply:
Or could we add `topologyInfo: Option[String]` in `Location`, like other fields `host/port/` which are from `BlockManagerId`?

`location.storageType.isReliable` is a good idea for "decommission".

Reply:
Based on discussion, if we are going to mirror everything in `BlockManagerId`, why not expose that directly with ability to subclass it and avoid `Location` altogether ?

Reply:
I think it (exposing `BlockManagerId`) depends on the use cases. We could probably expose it if `BlockManagerId` is generally satisfied all the common use cases(e.g., BlockManager, hdfs, file server). That being said, I think the concept of `BlockManagerId` is not suitable for every case. For example, each executor would be corresponding to a `BlockManagerId`.  But, in the case of hdfs, all the executors would only have one corresponding location indeed. 

Adding `Location` would be more flexible for users if they have specific requirements that `BlockManagerId` can't satisfy. Besides, I think it's safer for Spark as `BlockManagerId` is not only used for shuffle phase but also widely used in RDD cache.

Reply:
> Or could we add topologyInfo: Option[String] in Location, like other fields host/port/... which are from BlockManagerId?

Do you know other cases that need it?

Reply:
Uber Remote Shuffle Service uses topologyInfo to store shuffle server information. Since we kind of mirror things in `BlockManagerId`, thus suggest adding `topologyInfo` inside Location as well.

Reply:
I think in `Location` to specify these methods violates the abstraction. I know this convenient and avoids casting but still we should avoid meaningless methods. In a disaggregated storage solution the `executorId` has no value and probably host and port is not enough and never needed as separate entities but an URL like construct will be more useful (or something else but it is the responsibility of the specific subclass).